### PR TITLE
WIP: RST 364 sauce labs setup

### PR DESCRIPTION
**Pull request by AlexaVB**
https://wiki.saucelabs.com/display/DOCS/Instant+Selenium+Ruby+Tests#InstantSeleniumRubyTests-CodeSample

This is wip

JIRA https://tools.hmcts.net/jira/browse/RST-364

### CODE REVIEW

File: Local State [hunk 1]
```diff
@@ -0,0 +1 @@
+{"background_mode":{"enabled":false},"ssl":{"rev_checking":{"enabled":false}}}
```
Comment (garytaylor) [inline Local State]: Add to gitignore ?

File: config/saucelabs.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+username: aballantine
+authkey: c09c4ae6-c68f-419d-b09a-922404c5bb23
```
Comment (garytaylor) [inline config/saucelabs.yml]: Should this be in a public repo ?  Does  it not mean someone else could use our account ?

File: features/support/saucelabs/browsers.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (leoapost) [inline features/support/saucelabs/browsers.rb]: Is this file necessary? Shouldn't the `browser` method in the `SauceDriver` module be enough?

File: config/saucelabs.yml (hunk 2 above)
Comment (leoapost) [inline config/saucelabs.yml]: we can move saucelab's conf  in the `config/settings.yml`
e.g.
```
saucelabs:
  username: <%= ENV['SAUCELABS_USERNAME'] %>
  auth_key: <%= ENV['SAUCELABS_AUTH_KEY'] %>
```

then we can use env variable or the `.env` files to set the credentials

File: features/support/sauce_driver.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+module SauceDriver
+  class << self
+    def config
```
Comment (leoapost) [inline features/support/sauce_driver.rb]: We already get this functionality from the `config` gem.  Check my comment above about using the `Settings.saucelabs.something`.

File: features/support/sauce_driver.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      @config ||= YAML.load_file(Rails.root + 'config/saucelabs.yml')
+    end
+
+    def sauce_endpoint
```
Comment (leoapost) [inline features/support/sauce_driver.rb]: We could also move this in the settings file and access it with something like `Settings.saucelabs.on_demand_url`.
